shared/lib_financial.cpp

```diff
@@ -35,153 +35,244 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <limits>
 #include "lib_financial.h"
 
-using namespace libfin;
+#ifndef WIN32
+#include <float.h>
+#endif
+
+double libfin::min(double a, double b)
+{ // handle NaN
```

Comment:
Can you or @sjanzou expand upon this comment to include _why_ 0 is the correct answer when one of the inputs is NaN? My hunch based on reading the code is to make the cashflow readable, but it would be good to document that if true so future developers know whether to use this or std::min.

Reply:
@brtietz , yes - to make the cashflow readable and that std::min and max have screwy results based on if a NaN is the first element in the comparison - see https://stackoverflow.com/questions/72296440/how-to-prevent-stdmin-and-max-to-return-nan-if-the-first-element-of-the-array